src/rpc/request.cpp

```diff
@@ -82,7 +82,26 @@ static fs::path GetAuthCookieFile(bool temp=false)
 
 static bool g_generated_cookie = false;
 
-bool GenerateAuthCookie(std::string *cookie_out)
+static std::string perms_to_str(fs::perms p)
+{
+    char special = '-';
+    if ((p & fs::perms::set_uid)    != fs::perms::none) special = '4';
+    if ((p & fs::perms::set_gid)    != fs::perms::none) special = '2';
+    if ((p & fs::perms::sticky_bit) != fs::perms::none) special = '1';
```

Comment:
This won't work if multiple bits are set

Reply:
These bits seem to typically manifest by replacing the 'x' field with either 's' (setuid/setgid+executable), 'S' (setuid/setgid WITHOUT executable), 't' (sticky+exec), or 'T' (sticky NO exec)